### PR TITLE
CDSR-1196: Include account details

### PR DESCRIPTION
**Pull request by airudah2000**

### CODE REVIEW

File: app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala [hunk 1]
```diff
@@ -485,6 +482,36 @@ object DefaultClaimTransformerService {
...
+                postalCode = cd.postalCode,
+                telephoneNumber = cd.telephone,
+                faxNumber = None,
+                emailAddress = cd.emailAddress
+              )
+            }
+          )
+        })
+      case None                                      =>
+        Invalid(NonEmptyList.one("could not find account details"))
```
Comment (steve-edmans) [inline app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala:512]: Not having any account details specified is not an error. I don't think we should be failing validation in this case.

File: test/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerServiceSpec.scala [hunk 2]
```diff
@@ -893,7 +894,8 @@ class ClaimTransformerServiceSpec extends AnyWordSpec with Matchers with MockFac
         val contactDetails = None
         val contactAddress = None
 
-        val acc14                                        = getAcc14Response
+        val accountDetails                               = gen[AccountDetails].arbitrary.sample.map(List(_))
```
[the remark this answers is not in the document]
Comment (steve-edmans) [reply test/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerServiceSpec.scala:897]: You can also do `sample(List[AccountDetails])`

File: app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala (hunk 1 above)
Comment (airudah2000) [reply app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala]: Agree. I've changed this to return an `Option` instead.

File: test/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerServiceSpec.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (airudah2000) [reply test/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerServiceSpec.scala]: Changed to `List(sample[AccountDetails])`

**Review by aneksamun — APPROVED** [review]

**Pull request merged**